MsWheaPkg/MsWheaReport/MsWheaEarlyStorageMgr.c

```diff
@@ -229,17 +228,9 @@ MsWheaESContentChangeChecksumHelper (
   DEBUG ((DEBUG_INFO, "Calculate sum content helper...\n"));
 
   MsWheaESReadHeader (&Header);
-  Sum16 = (UINT16)(0x10000 - Header.Checksum);
-
-  // Update length field of header, update checksum accordingly
-  Sum16               = (Sum16 - (UINT16)(Header.ActiveRange & MAX_UINT16) - (UINT16)((Header.ActiveRange >> 16) & MAX_UINT16));
   Header.ActiveRange += (UINT32)Length;
-  Sum16               = (Sum16 + (UINT16)(Header.ActiveRange & MAX_UINT16) + (UINT16)((Header.ActiveRange >> 16) & MAX_UINT16));
-
-  // Calculate updated chunk only
-  Sum16           = Sum16 + CalculateSum16 (Buffer, Length);
-  Header.Checksum = (UINT16)(0x10000 - Sum16);
-
+  MsWheaESCalculateChecksum16 (&Header, &Sum16);
+  Header.Checksum = Sum16;

```

Comment:
nit: Do you need to 0 this field?

Reply:
The MsWheaESCalculateChecksum16() zeros the checksum before doing the calculation. Is that what you mean?

Reply:
yep, thanks!